### PR TITLE
fix interface for merge

### PR DESCRIPTION
**Pull request by schieb**
fixes bug in kwargs.pop() when common_history is not provided

### CODE REVIEW

**Comment by schieb** [comment]
Resubmission of angr/pull/410

**Comment by zardus** [comment]
It looks like the fact that the function used to take an `*args` and now takes an explicit list is causing failures in other places...

**Comment by schieb** [comment]
Squashed a commit that fixes that issue. Not sure how to trigger the CI that failed last time.

**Comment by zardus** [comment]
The CI should automatically run, but this doesn't seem to be happening...

**Comment by zardus** [comment]
My only guess about the CI is that it's holding out for the comment I made on on path.py to be resolved?

File: angr/path.py [hunk 1]
```diff
@@ -571,21 +571,18 @@ def _add_stack_region_mapping(self, state, sp=None, ip=None):
...
         :rtype: Path
         """
 
-        common_history = kwargs.pop('common_history')
-        if len(kwargs) != 0:
-            raise ValueError("invalid arguments: %s" % kwargs.keys())
-
+        all_paths = other_paths + [self]
         if len(set(( o.addr for o in all_paths))) != 1:
             raise AngrPathError("Unable to merge paths.")
```
Comment (zardus) [inline angr/path.py]: Resolving this, though I still think we should do it :-)

**Comment by zardus** [comment]
Woo, looks like travis woke up after that push!

**Comment by zardus** [comment]
Thanks for the contribution!

**Pull request merged**